doc/devices/drivers/gpio/gpio.rst

```diff
@@ -0,0 +1,294 @@
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_DRIVE_STRENGTH_7`
+
+**GPIO_DS_DISCONNECT_LOW**
+    Disconnect pin when GPIO pin output is low.
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_DRIVE_OPEN_SOURCE`
+
+**GPIO_DS_DISCONNECT_HIGH**
+    Disconnect pin when GPIO pin output is high.
+
+    PINCTRL pin configuration property:
+
+    - :c:macro:`PINCTRL_CONFIG_DRIVE_OPEN_DRAIN`
+
+Signalling Configuration
+========================
+
+Pin signalling can be configured by the general GPIO configuration interface.
+
+:c:func:`gpio_pin_configure`
+    Set the configuration of a pin.
+
+A generic set of GPIO port pin signalling properties is defined in
+:file:`include/gpio.h`. A GPIO driver may only support a subset of the
+signalling properties.
+
+**GPIO_INT**
+    GPIO pin to trigger interrupt.
+
+**GPIO_INT_ACTIVE_LOW**
+    GPIO pin trigger on level low or falling edge.
+
+**GPIO_INT_ACTIVE_HIGH**
+    GPIO pin trigger on level high or rising edge.
+
+**GPIO_INT_CLOCK_SYNC**
+    GPIO pin trigger to be synchronized to clock pulses.
+
+**GPIO_INT_DEBOUNCE**
+    Enable GPIO pin trigger debounce.
+
+**GPIO_INT_LEVEL**
+    Do Level trigger.
+
+**GPIO_INT_EDGE**
+    Do Edge trigger.
+
+**GPIO_INT_DOUBLE_EDGE**
+    Interrupt triggers on both rising and falling edge.
+
+
+GPIO device tree support
+************************
+
+Nodes
+=====
+
+GPIO Controller Node
+--------------------
+
+The GPIO device is defined by a GPIO controller node.
+
+Required properties are:
+
+    - gpio-controller: Indicates this device is a GPIO controller
+    - compatible:
+    - label: Should be a name string
+
+
+Interaction with the PINCTRL device driver
+==========================================
+
+To enable a pin controller to report the pins that are associated to a GPIO device
+the gpio-ranges property shall be set in the device tree. Also the label property
+shall be set and used as device name.
+
+The gpio-ranges property is of the form:
+
+.. code-block:: text
+
+    gpio-ranges = <&[pin controller] [pin number in GPIO pin number space]
+                                     [pin number in pin controller number space]
+                                     [number of pins]>,
+                                     /* ... */
+                                     ;
+
+A GPIO port definition in the device tree may look like:
+
+.. code-block:: DTS
+
+    pinctrl: pin-controller@48000000 {
+        pin-controller;
+        /* ... */
+    };
+
+    gpioa: gpio@40020000 {
+        gpio-controller;
+        compatible = "st,stm32-gpio-pinctrl";
+        #gpio-cells = <2>;
+        reg = <0x40020000 0x400>;
+        label = "STM32_GPIOA";
+        gpio-ranges = <&pinctrl GPIO_PORT_PIN0 PINCTRL_STM32_PINA0 16>;
+    };
+
+
+GPIO device driver implementation
+*********************************
+
+Driver configuration information comes from three sources:
+    - autoconf.h (Kconfig)
```

Comment:
Add a blank line before the first list item and outdent them all so the ``-`` lines up with the ``D`` in ``Driver``.